### PR TITLE
Watch point tests

### PR DESCRIPTION
**Pull request by KyleDavisSA**
## Main changes of this PR
This PR aims to add a nearest neighbor test to the watch points, as well as parallel tests with 1 vertex on each rank.

## Motivation and additional information
closes https://github.com/precice/precice/issues/781

## Author's checklist

* [ ] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I ran `make format` to ensure everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?

### CODE REVIEW

**Comment by KyleDavisSA** [comment]
The test does not perform parallel tests yet. From `PetRadialBasisFctMappingTests` they use the `TestContext`, however this is not working yet. The idea is to add another argument, e.g. ` bool inParallel`, and this would trigger the number of ranks to run the test on.

**Comment by fsimonis** [comment]
I had to rebase on develop and fix some naming to get this working.

File: src/precice/tests/WatchPointTest.cpp [hunk 1]
```diff
@@ -38,122 +39,238 @@ std::vector<double> readDoublesFromTXTFile(std::string const &filename, int skip
...
+{
+  PRECICE_TEST(1_rank);
+  bool withEdge           = false;
+  auto watchPointPosition = std::vector<double>{-0.5, 0.6};
+  auto expected           = std::vector<double>{
+      0.0, 0.0, 1.0, 2.0, 3.0, 4.0,
+      1.0, 0.0, 1.0, 3.0, 4.0, 5.0,
+      2.0, 0.0, 1.0, 3.0, 4.0, 5.0};
+  testWatchPoint(context, withEdge, watchPointPosition, expected);
+}
```
Comment (fsimonis) [inline src/precice/tests/WatchPointTest.cpp:192]:
```suggestion
}

```

File: src/precice/tests/WatchPointTest.cpp [hunk 2]
```diff
@@ -38,122 +39,238 @@ std::vector<double> readDoublesFromTXTFile(std::string const &filename, int skip
...
+{
+  PRECICE_TEST(1_rank);
+  bool withEdge           = false;
+  auto watchPointPosition = std::vector<double>{0.0, 1.6};
+  auto expected           = std::vector<double>{
+      0.0, 0.0, 2.0, 3.0, 5.0, 6.0,
+      1.0, 0.0, 2.0, 4.0, 6.0, 7.0,
+      2.0, 0.0, 2.0, 4.0, 6.0, 7.0};
+  testWatchPoint(context, withEdge, watchPointPosition, expected);
+}
```
Comment (fsimonis) [inline src/precice/tests/WatchPointTest.cpp:215]:
```suggestion
}

```

**Review by fsimonis — APPROVED** [review]

**Comment by fsimonis** [comment]
Looks good. Let's just wait and see why the CI failed on MacOS.

**Pull request merged**